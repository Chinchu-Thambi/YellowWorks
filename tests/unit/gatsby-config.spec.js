/* globals describe, it, expect */

import { morph } from 'mock-env';

morph(() => {
  // eslint-disable-next-line global-require
  const config = require('../../gatsby-config');

  describe('gatsby config', () => {
    it('should not change by accident', () => {
      expect(config).toMatchSnapshot();
    });
  });
}, {
  CONTENTFUL_SPACE_ID: 'mock-space-id',
  CONTENTFUL_ACCESS_TOKEN: 'mock-access-token',
  CONTENTFUL_PREVIEW_TOKEN: 'mock-preview-token',
  CONTENTFUL_HOST: 'mock-host',
  SITE_URL: 'mock-site-url',
  GATSBY_HUBSPOT_ACCOUNT: 'mock-hubspot-account',
  BIGCOMMERCE_CLIENT_ID: 'mock-bigcommerce-clientid',
  BIGCOMMERCE_CLIENT_SECRET: 'mock-bigcommerce-secret',
  BIGCOMMERCE_TOKEN: 'mock-bigcommerce-token',
  BIGCOMMERCE_STORE_HASH: 'mock-bigcommerce-store-hash',
  SEGMENT_KEY: 'mock-segment-key',
  DATA_DOG_KEY: 'mock-datadog-key',
  ROBOTS_TXT_TEMPLATE_ENV: 'mock-nonproduction',
  IAM_PRODUCT_PLATFORM_ENDPOINT: 'http://mock-product-platform/',
});
