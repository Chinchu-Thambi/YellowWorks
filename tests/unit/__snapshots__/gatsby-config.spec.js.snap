// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby config should not change by accident 1`] = `
Object {
  "developMiddleware": [Function],
  "plugins": Array [
    Object {
      "options": Object {
        "displayName": false,
      },
      "resolve": "gatsby-plugin-styled-components",
    },
    "gatsby-plugin-react-helmet",
    Object {
      "options": Object {
        "background_color": "#FFFFFF",
        "display": "standalone",
        "icon": "src/assets/logos/y-favicon.png",
        "name": "Yellow NZ",
        "short_name": "Yellow NZ",
        "start_url": "/",
        "theme_color": "#1F2A44",
      },
      "resolve": "gatsby-plugin-manifest",
    },
    Object {
      "options": Object {
        "exclude": Array [
          "/my-yellow/",
          "/my-yellow/*",
          "/our-products/search-ads/onboarding/",
          "/our-products/search-ads/onboarding/*",
          "/example/",
          "/example/*",
          "/demo/",
          "/demo/*",
        ],
        "output": "sitemap_statics.xml",
        "serialize": [Function],
      },
      "resolve": "gatsby-plugin-sitemap",
    },
    "gatsby-plugin-offline",
    Object {
      "options": Object {
        "postCssPlugins": Array [
          Processor {
            "plugins": Array [
              [Function],
              [Function],
            ],
            "postcssPlugin": "tailwind",
            "postcssVersion": "7.0.21",
            "version": "7.0.21",
          },
          [Function],
        ],
      },
      "resolve": "gatsby-plugin-postcss",
    },
    Object {
      "options": Object {
        "display": "swap",
        "fonts": Array [
          "lato:400,400i,700",
          "montserrat:400,700",
        ],
      },
      "resolve": "gatsby-plugin-google-fonts",
    },
    "gatsby-plugin-no-sourcemaps",
    Object {
      "resolve": "gatsby-plugin-react-axe",
    },
    Object {
      "options": Object {
        "exclude": /\\(node_modules\\|\\.cache\\|public\\)/,
        "options": Object {
          "emitWarning": true,
          "failOnError": true,
        },
        "stages": Array [
          "develop",
          "build-javascript",
        ],
        "test": /\\\\\\.js\\$\\|\\\\\\.jsx\\$/,
      },
      "resolve": "gatsby-plugin-eslint",
    },
    Object {
      "options": Object {
        "accessToken": "mock-preview-token",
        "host": "mock-host",
        "spaceId": "mock-space-id",
      },
      "resolve": "gatsby-source-contentful",
    },
    Object {
      "options": Object {
        "fetch": [Function],
        "fieldName": "productPlatform",
        "typeName": "ProductPlatform",
        "url": "http://mock-product-platform/",
      },
      "resolve": "gatsby-source-graphql",
    },
    Object {
      "options": Object {
        "trackingCode": "mock-hubspot-account",
      },
      "resolve": "gatsby-plugin-hubspot",
    },
    Object {
      "options": Object {
        "devKey": "mock-segment-key",
        "prodKey": "mock-segment-key",
      },
      "resolve": "gatsby-plugin-segment-js",
    },
    Object {
      "options": Object {
        "disable": false,
        "onLoad": "() => DD_LOGS.init({
            clientToken: 'mock-datadog-key',
            forwardErrorsToLogs: true,
        })",
        "src": "https://www.datadoghq-browser-agent.com/datadog-logs-us.js",
      },
      "resolve": "gatsby-plugin-load-script",
    },
    Object {
      "options": Object {
        "async": true,
      },
      "resolve": "gatsby-plugin-stripe",
    },
    Object {
      "options": Object {
        "crossorigin": true,
        "hash": "sha512",
      },
      "resolve": "gatsby-plugin-sri",
    },
    Object {
      "options": Object {
        "env": Object {
          "development": Object {
            "policy": Array [
              Object {
                "disallow": Array [
                  "/api/",
                  "/services/",
                  "/user/",
                  "reset/",
                  "listing/",
                  "review/",
                  "designs/",
                  "/*?_bc_fsnf=1*",
                  "/*&_bc_fsnf=1*",
                ],
                "userAgent": "*",
              },
              Object {
                "crawlDelay": 5,
                "disallow": Array [
                  "/api/",
                  "/services/",
                  "/user/",
                  "/reset/",
                  "/listing/",
                  "/review/",
                  "/designs/",
                ],
                "userAgent": "bingbot",
              },
              Object {
                "disallow": Array [
                  "/account.php",
                  "/cart.php",
                  "/checkout.php",
                  "/finishorder.php",
                  "/login.php",
                  "/orderstatus.php",
                  "/postreview.php",
                  "/productimage.php",
                  "/productupdates.php",
                  "/remote.php",
                  "/search.php",
                  "/viewfile.php",
                  "/wishlist.php",
                  "/admin/",
                  "/__socialshop/ ",
                ],
                "userAgent": "AdsBot-Google",
              },
              Object {
                "allow": "/feeds/*",
                "userAgent": "google-xrawler",
              },
            ],
          },
          "nonproduction": Object {
            "policy": Array [
              Object {
                "disallow": Array [
                  "/",
                ],
                "userAgent": "*",
              },
            ],
          },
          "production": Object {
            "policy": Array [
              Object {
                "disallow": Array [
                  "/api/",
                  "/services/",
                  "/user/",
                  "reset/",
                  "listing/",
                  "review/",
                  "designs/",
                  "/*?_bc_fsnf=1*",
                  "/*&_bc_fsnf=1*",
                ],
                "userAgent": "*",
              },
              Object {
                "crawlDelay": 5,
                "disallow": Array [
                  "/api/",
                  "/services/",
                  "/user/",
                  "/reset/",
                  "/listing/",
                  "/review/",
                  "/designs/",
                ],
                "userAgent": "bingbot",
              },
              Object {
                "disallow": Array [
                  "/account.php",
                  "/cart.php",
                  "/checkout.php",
                  "/finishorder.php",
                  "/login.php",
                  "/orderstatus.php",
                  "/postreview.php",
                  "/productimage.php",
                  "/productupdates.php",
                  "/remote.php",
                  "/search.php",
                  "/viewfile.php",
                  "/wishlist.php",
                  "/admin/",
                  "/__socialshop/ ",
                ],
                "userAgent": "AdsBot-Google",
              },
              Object {
                "allow": "/feeds/*",
                "userAgent": "google-xrawler",
              },
            ],
          },
        },
        "host": "mock-site-url",
        "resolveEnv": [Function],
        "sitemap": "mock-site-url/sitemap.xml",
      },
      "resolve": "gatsby-plugin-robots-txt",
    },
  ],
  "siteMetadata": Object {
    "siteUrl": "mock-site-url",
  },
}
`;
